app/controllers/book/index.js

```diff
@@ -29,15 +30,11 @@ export default Controller.extend({
     },
 
     search({ target }) {
-      debounce(() => {
-        this.set("search", target.value);
-      }, 1000);
+      debounce(() => set(this, "search", target.value), 1000);
```

Comment:
Это тоже самое, в `debounce` всё равно каждый раз передаётся новая функция.
```suggestion
      debounce(this, this.set, 'search', target.value, 1000);
```
Так будет вызываться метод `set` с контекстом `this`, и получать `'search'` и `target.value` в параметры.